pkg/oci/containerize_test.go

```diff
@@ -0,0 +1,73 @@
+package oci
+
+import (
+	"os"
+	"path/filepath"
+	"runtime"
+	"testing"
+)
+
+// Test_validatedLinkTaarget ensures that the function disallows
+// links which are absolute or refer to targets outside the given root, in
+// addition to the basic job of returning the value of reading the link.
+func Test_validatedLinkTarget(t *testing.T) {
+	root := "testdata/test-links"
+
+	allRuntimes := []string{} // optional list of runtimes a test applies to
+
+	testAppliesToCurrentRuntime := func(testRuntimes []string) bool {
+		if len(testRuntimes) == 0 {
+			return true // no filter defined.
+		}
+		for _, r := range testRuntimes {
+			if runtime.GOOS == r {
+				return true // filter defined, and current is defined.
+			}
+		}
+		return false // filter defined; current not in set.
+	}
+
+	tests := []struct {
+		path     string   // path of the file within test project root
+		valid    bool     // If it should be considered valid
+		target   string   // optional test of the returned value (target)
+		name     string   // descriptive name of the test
+		runtimes []string // only apply this test to the given runtime(s)
+	}{
+		{"absoluteLink", false, "", "disallow absolute-path links on linux", []string{"linux"}},
+		{"absoluteLinkWindows", false, "", "disallow absolute-path links on windows", []string{"windows"}},
```

Comment:
I am confused. I still do not understand why this is written like this.

Why not just have single test:
```go
absoluteLink := "absoluteLink"
if runtime.GOOS == "windows" {
    absoluteLink = "absoluteLinkWindows"
}
tests := []struct {
		path     string   // path of the file within test project root
		valid    bool     // If it should be considered valid
		target   string   // optional test of the returned value (target)
		name     string   // descriptive name of the test
	}{
		{absoluteLink, false, "", "disallow absolute-path links on linux"},
```

Reply:
Because this way we can simply list all the tests and optionally provide a set of platforms for which they apply.  With your example, there's a stack of if-statements above override tests implementaitons.

With this method, all tests are listed in the same way, and those with specific OS conditions can clearly state as such in their name.

This is stylistic preference favoring the simplicity of the test slice, not an architectural or logical error.

Reply:
OK, personally do not find it more readable, but I won't argue about it.